sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -271,7 +271,15 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
         ignoreIfExists)
     } else {
       val tableWithDataSourceProps = tableDefinition.copy(
-        schema = hiveCompatibleSchema,
+        schema = hiveCompatibleSchema match {
+          // Spark-created views do not have to be Hive compatible. If the data type is not
+          // Hive compatible, we can set schema to empty so that Spark can still read this
+          // view as the schema is also encoded in the table properties.
+          case schema if schema.exists(f => !isHiveCompatibleDataType(f.dataType)) &&
+              tableDefinition.tableType == CatalogTableType.VIEW =>
```

Comment:
Switch the order of 2 guardians?

Reply:
Does it look https://github.com/apache/spark/pull/50367/files#diff-81dcb1eea508dd5886eadedce48c89169b812224020a75d38e66aed0ee71a039R296-R302 redundant now?